File: Project phase Problem solving part/41-first-missing-positive/41-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        
        for i in range(len(nums)):
            if nums[i] <= 0:
                nums[i] = len(nums) + 1
        
        for i in range(len(nums)):
            if abs(nums[i]) <= len(nums):
                val = abs(nums[i])
                if nums[val - 1] > 0:
                    nums[val - 1] *= -1
        
        for i in range(len(nums)):
            if nums[i] > 0:
                return i + 1
        
        return len(nums) + 1

File: 41-first-missing-positive/41-first-missing-positive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var firstMissingPositive = function(nums) {
    
    for(let i = 0; i < nums.length; i++){
        if (nums[i] <= 0){
            nums[i] = nums.length + 1;
        }
    }
    
    for(let i = 0; i < nums.length; i++){
        if(Math.abs(nums[i]) <= nums.length){
            var val = Math.abs(nums[i]);
            if (nums[val - 1] > 0){
                
                nums[val - 1] *= -1;
            }
        } 
    }
    
    for(let i = 0; i < nums.length; i++){
        if (nums[i] > 0){
            return i + 1;
        }
    }
    
    return nums.length + 1
};
        
